railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -184,6 +187,10 @@ def initialize(*args)
         if !options[:skip_active_record] && !DATABASES.include?(options[:database])
           raise Error, "Invalid value for --database option. Supported for preconfiguration are: #{DATABASES.join(", ")}."
         end
+
+        # Force sprockets to be skipped when generating http only app.
+        # Can't modify options hash as it's frozen by default.
+        self.options = options.merge(skip_sprockets: true, skip_javascript: true).freeze if options[:api]
```

Comment:
This is implemented in this way taking advantage of existent conditional logic in templates.

Examples: https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt#L29
https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/app/templates/config/environments/production.rb.tt#L21

I think it's better to have this done here, so we can avoid having more conditionals in the templates.

Thoughts?